src/CtaLink/CtaLink.js

```diff
@@ -0,0 +1,268 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ctaLinkIcons, ctaLinkDisplay, ctaLinkColors, ctaLinkAnimations, ctaLinkButtonSizes, ctaLinkButtonVariants } from './CtaLink.levers';
+import Icon from 'react-hero-icon';
+import clsxd from 'clsx-dedupe';
+
+/**
+ * CTA Link Component
+ *
+ */
+export const CtaLink = ({ classes = {}, text, srText, display, link, color, isButton, variant, size, icon, animate, ref, ...props }) => {
+  // Defaults & Variables.
+  // ---------------------------------------------------------------------------
+  const levers = {};
+
+  // Levers
+  // ---------------------------------------------------------------------------
+
+  // display
+  if (display && ctaLinkDisplay.includes(display)) {
+    switch (display) {
+      case 'inline-block':
+        levers.display = 'su-inline-block';
+        break;
+
+      case 'block':
+        levers.display = 'su-block';
+        break;
+
+      case 'flex':
+        levers.display = 'su-flex';
+        break;
+    }
+  }
+
+  // color
+  if (color && ctaLinkColors.includes(color) && !isButton) {
+    switch (color) {
+      case 'red':
+        levers.color = 'su-text-digital-red hocus:su-text-black';
+        break;
+
+      case 'blue':
+        levers.color = 'su-text-digital-blue hocus:su-text-black';
+        break;
+
+      case 'white':
+        levers.color = 'su-text-white hocus:su-text-white';
+        break;
+    }
+  }
+
+  // isButton
+  if (isButton) {
+    levers.isButton = 'su-cta-btn su-font-regular su-leading-none';
+  }
+  else {
+    levers.isButton = 'su-cta-link su-text-19 md:su-text-20';
+  }
+
+  // variant
+  if (variant && ctaLinkButtonVariants.includes(variant) && isButton) {
+    switch (variant) {
+      case 'solid':
+        levers.variant = 'su-bg-digital-red hocus:su-bg-archway-dark su-text-white hocus:su-text-white su-border-2 su-border-digital-red su-border-solid hover:su-border-black focus:su-border-black';
+        break;
+
+      case 'outline':
+        levers.variant = 'su-bg-transparent hocus:su-bg-transparent su-text-digital-red hocus:su-text-black su-border-2 su-border-digital-red su-border-solid hover:su-border-black focus:su-border-black';
+        break;
+
+      case 'ghost':
+        levers.variant = 'su-bg-transparent su-text-white hocus:su-text-white su-border-2 su-border-white su-border-solid';
+        break;
+    }
+  }
+
+  // size
+  if (size && ctaLinkButtonSizes.includes(size) && isButton) {
```

Comment:
Perhaps the button link style should be its own component? There are quite a few conditionals on  the isButton prop.

Reply:
😂  I started out having the button link its own component, then I realized that there were quite a bit of shared code (the icons and animation stuff), so I decided to combine them. I did make 2 separate "stories" files though so it's a bit more manageable.

Want to make a decision?

Reply:
> 😂 I started out having the button link its own component, then I realized that there were quite a bit of shared code (the icons and animation stuff), so I decided to combine them. I did make 2 separate "stories" files though so it's a bit more manageable.
> 
> Want to make a decision?

Abstraction, and the right amount, can be tough. If you have shared items amongst multiple objects you can pull them out into a higher parent if you are using inheritance but alas we are not. We tend to import shared libraries in most of our work. Do a little reading (I'll do the same) at different abstraction techniques in Javascript/React, and we can come back to the design on this.

Reply:
Yvonne and I talked about this higher-level issue in our 1:1 yesterday. One of the fundamental challenges that we face in SWS is writing code that scales; i.e., abstraction. Thanks to you both for putting the thought and discussion into this.

Reply:
Yeah agree about finding the right balance. I'll poke around and see if I can find a good example of sharing common options.

Reply:
After reading that article you shared, and thinking about decanter 6 experience, I think I'm going to separate out CTA Link and CTA Link Button into 2 different components so it doesn't have a crap load of props, and it will be easier for future maintenance. I might or might not experiment with sharing some of the levers code (eg, icons) in this PR. Let me know what you think.

Reply:
I think that is a good approach. Every article I read argues for composition over configuration and the more we can do that in this repo the better. We are knowingly going against recommended practices in order to support our design team and would do good by ourselves to try to keep to some best practices where we can. More components and shared levers might give us a chance to grow and maintain this thing before it becomes too complicated and untestable.

Reply:
Sounds good 😄  I'll try my best to keep it simple stupid.